services/twilioService.js

```diff
@@ -0,0 +1,225 @@
+import { Client as TwilioChatClient } from 'twilio-chat';
+import { AsyncStorage } from 'react-native';
+import { CONFIG } from '../config.js';
+import MessageService from './messageService.js';
+
+const API_URL = CONFIG.ENV === 'PROD' ? CONFIG.API_URL : CONFIG.TEST_API_URL;
+
+class TwilioService {
+  constructor(hmToken, contacts, messagesCallback) {
+    this.hmToken = hmToken;
+    this.id = `${JSON.parse(hmToken)._id}`;
+
+    this.chatClient = null;
+    this.messageService = new MessageService(messagesCallback);
+
+    // Map user_id to chat channel object
+    this.channels = {};
+    this.contacts = [contacts];
+
+    this.startup(contacts);
+  }
+
+  async startup(contacts) {
+    const success = await this.loadTwilioClient();
+    if (success) {
+      // Initialize all channels
+      const doneInitChannels = await this.initAllChannels(contacts);
+      if (doneInitChannels) {
+        await this.getAllMessages();
+      } else {
+        // TODO: something bad happened
```

Comment:
We should really look into getting that sentry PR merged, or else these are just gonna pile up

Reply:
Yeah agreed. I'll take on that integration once this PR is merged to master.